cloud-manager/pkg/services/kubernetes_service.go

```diff
@@ -0,0 +1,181 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/mattermost/mattermost-cloud-monitoring/cloud-manager/pkg/providers"
+	k8sdrain "github.com/openshift/kubernetes-drain"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const PodUnschedulable = "node.kubernetes.io/unschedulable"
+
+type Service struct {
+	contextName string
+	client      *kubernetes.Clientset
+	provider    providers.Provider
+}
+
+func NewKubernetesService(contextName string, provider providers.Provider) (*Service, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", defaultPath())
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Failed to build k8s config: %s", err.Error()))
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Failed to create k8s client: %s", err.Error()))
+	}
+
+	return &Service{contextName: contextName, client: client, provider: provider}, nil
+}
+
+func defaultPath() string {
+	return filepath.Join(homeDir(), ".kube", "config")
+}
+
+func homeDir() string {
+	if h := os.Getenv("HOME"); h != "" {
+		return h
+	}
+	return os.Getenv("USERPROFILE") // windows
+}
+
+func (s *Service) Drain(force, ignoreDaemonsets, deleteLocalData bool, namespace string, gracePeriodSeconds int, timeout time.Duration, forceTermination bool) error {
+	nodeList, err := s.client.CoreV1().Nodes().List(metav1.ListOptions{})
+
+	if err != nil {
+		log.Println(fmt.Sprintf("Failed to fetch k8s nodes due to %s", err.Error()))
```

Comment:
Instead of printing every error and then returning the error, we could just print the returned error in one spot.